graph2tac/tfgnn/predict.py

```diff
@@ -425,18 +358,82 @@ def _expand_arguments_logits(cls,
 
     def _inference_model(self, tactic_expand_bound: int) -> tf.keras.Model:
         if tactic_expand_bound not in self._inference_model_cache.keys():
-            inference_model = self.prediction_task.create_inference_model(tactic_expand_bound=tactic_expand_bound,
+            inference_model_bare = self.prediction_task.create_inference_model(tactic_expand_bound=tactic_expand_bound,
                                                                           graph_constants=self.graph_constants)
+            tactic_mask_spec = tf.TensorSpec(shape=(self.graph_constants.tactic_num,), dtype=bool)
+            @tf.function(input_signature = (LoaderProofstateSpec, tactic_mask_spec))
+            def inference_model(state, tactic_mask):
+                graph_tensor_single = self._make_proofstate_graph_tensor(state)
+                graph_tensor_stacked = stack_graph_tensors([graph_tensor_single])
+                inference_output = inference_model_bare({
+                    self.prediction_task.PROOFSTATE_GRAPH: graph_tensor_stacked,
+                    self.prediction_task.TACTIC_MASK: tf.expand_dims(tactic_mask, axis=0),
+                })
+                return inference_output
             self._inference_model_cache[tactic_expand_bound] = inference_model
         return self._inference_model_cache[tactic_expand_bound]
 
+    def _tactic_mask_from_allowed_model_tactics(self, allowed_model_tactics: Optional[Iterable[int]]) -> tf.Tensor:
```

Comment:
I'm worried about this code causing problems since it isn't compiled.  Did this already exist in the code base previously?  If not, can we have it produce numpy arrays instead?

Reply:
Actually, it was already in the code.  I guess it is fine.